File: ResourceManager/Services/Abstractions/IRepositoryDataService.cs
using ResourceManager.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ResourceManager.Services.Abstractions
{
    public interface IRepositoryDataService : IDataService<Repository>
    {
        Task<int> UpdateTags(Repository repo, IList<Tag> tags);
    }
}

File: ResourceManager/Services/Abstractions/IRepositoryDataService.vb
Imports ResourceManager.Models
Imports System.Collections.Generic
Imports System.Threading.Tasks

Namespace ResourceManager.Services.Abstractions
    Public Interface IRepositoryDataService
        Inherits IDataService(Of Repository)
        Function UpdateTags(ByVal repo As Repository, ByVal tags As IList(Of Tag)) As Task(Of Integer)
    End Interface
End Namespace
